File: InterviewCakeProblems2/topScores.js
function sortScores(unsortedScores, highestPossibleScore) {
    let countingArr = new Array(highestPossibleScore + 1).fill(0);
    let sortedScores = [];

    for (let i = 0; i < unsortedScores.length; i++) {
        let currentScore = unsortedScores[i];
        countingArr[currentScore]++;
    }
    
    for (let i = countingArr.length - 1; i >= 0; i--) {
        let scoreFreq = countingArr[i];
        if (scoreFreq > 0) {
            while (scoreFreq > 0) {
                sortedScores.push(i);
                scoreFreq--;
            }
        }
    }
    return sortedScores;

};

const unsortedScores = [37, 89, 41, 65, 91, 53];
const unsortedScores2 = [37, 89, 41, 65, 37, 88, 100, 7, 21, 20, 10, 91, 53];
const HIGHEST_POSSIBLE_SCORE = 100;

console.log(sortScores(unsortedScores, HIGHEST_POSSIBLE_SCORE));
console.log(sortScores(unsortedScores2, HIGHEST_POSSIBLE_SCORE));

File: InterviewCakeProblems2/top_scores.rb
def sort_scores(unsorted_scores, highest_possible_score)
    counting_arr = Array.new(highest_possible_score + 1){0}
    sorted_scores = [];

    unsorted_scores.each do |score|
        counting_arr[score] += 1
    end
    
    i = counting_arr.length - 1

    while i >= 0
        score_freq = counting_arr[i]
        while score_freq > 0
            sorted_scores.append(i)
            score_freq -= 1
        end
        i -= 1
    end
    sorted_scores
end

unsorted_scores = [37, 89, 41, 65, 91, 53];
unsorted_scores2 = [37, 89, 41, 65, 37, 88, 100, 7, 21, 20, 10, 91, 53];
highest_possible_score = 100;

p (sort_scores(unsorted_scores, highest_possible_score));
p (sort_scores(unsorted_scores2, highest_possible_score));